docs/guides/javascript.rst

```diff
@@ -339,6 +339,30 @@ Calls an Elgg action with the data passed. This handles outputting of system mes
       }
    });
 
+Module elgg/Spinner
+~~~~~~~~~~~~~~~~~~~
+
+The ``elgg/Spinner`` module can be used to create an Ajax loading indicator fixed to the top of the window.
+
+.. code:: js
+
+   define(function (require) {
+      var Spinner = require('elgg/Spinner');
+
+      var spinner = new Spinner;
+      spinner.start();
+      elgg.action('friend/add', {
+          success: function (json) {
+             // ...
+          },
+          complete: function () {
+             spinner.stop();
+          }
+      });
+   });
+
+**Note:** The indicator is only removed once all Spinner instances are stopped.
+
```

Comment:
I changed it to use beforeSend, and switched this to reference the instance methods directly (also added a test to verify we can call these without `this` context)